### PR TITLE
BPI-129 - Use channel id in facet rather that channel name

### PR DESCRIPTION
**Pull request by rimi-itk**

### CODE REVIEW

File: src/Bpi/ApiBundle/Controller/ChannelController.php [hunk 1]
```diff
@@ -481,7 +481,7 @@ public function addNodeToChannelAction()
         $dm->flush();
 
         $facetRepository = $this->getRepository('BpiApiBundle:Entity\Facet');
-        $facetRepository->addChannelToFacet($channel->getChannelName(), $params['nodes']);
+        $facetRepository->addChannelToFacet($channel->getId(), $params['nodes']);
```
Comment (makcumm) [inline src/Bpi/ApiBundle/Controller/ChannelController.php:484]: I suggest to change method arguments names.
https://github.com/inleadmedia/rest-api/blob/develop/src%2FBpi%2FApiBundle%2FDomain%2FRepository%2FFacetRepository.php#L298

File: src/Bpi/ApiBundle/Controller/ChannelController.php [hunk 2]
```diff
@@ -579,7 +579,7 @@ public function removeNodeFromChannelAction()
         $dm->flush();
 
         $facetRepository = $this->getRepository('BpiApiBundle:Entity\Facet');
-        $facetRepository->removeChannelFromFacet($channel->getChannelName(), $params['nodes']);
+        $facetRepository->removeChannelFromFacet($channel->getId(), $params['nodes']);
```
Comment (makcumm) [inline src/Bpi/ApiBundle/Controller/ChannelController.php:582]: Same here.
https://github.com/inleadmedia/rest-api/blob/develop/src%2FBpi%2FApiBundle%2FDomain%2FRepository%2FFacetRepository.php#L322

**Comment by makcumm** [comment]
Consider that this change will affect rest-api response in listing of channel facet, instead of channel names you will get channel id's.

File: src/Bpi/ApiBundle/Controller/ChannelController.php (hunk 1 above)
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Controller/ChannelController.php:484]: Good point. I will change the argument name.

File: src/Bpi/ApiBundle/Controller/ChannelController.php (hunk 2 above)
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Controller/ChannelController.php:582]: Good point. I will change the argument name.

**Comment by rimi-itk** [comment]
I have added channel names (using title attribute) to facets in rest API.

File: src/Bpi/ApiBundle/Controller/RestController.php [hunk 3]
```diff
@@ -280,6 +280,13 @@ function ($e) use ($document, $router) {
                         $term['count'],
                         $term['agencyName']
                     );
+                } else if (isset($term['count'])) {
```
Comment (makcumm) [inline src/Bpi/ApiBundle/Controller/RestController.php]: elseif should be wrote together.

File: src/Bpi/ApiBundle/Domain/Repository/FacetRepository.php [hunk 4]
```diff
@@ -172,6 +172,10 @@ function(key, values) {
                 $agency = $this->dm->getRepository('BpiApiBundle:Aggregate\Agency')->loadUserByUsername($facet['_id']['facetValue']);
                 $facets['agency_id'][$facet['_id']['facetValue']]['agencyName'] = $agency->getName();
                 $facets['agency_id'][$facet['_id']['facetValue']]['count'] = $facet['value'];
+            } else if ($facet['_id']['facetName'] == 'channels') {
```
Comment (makcumm) [inline src/Bpi/ApiBundle/Domain/Repository/FacetRepository.php]: elseif should be wrote together.

File: src/Bpi/ApiBundle/Controller/RestController.php (hunk 3 above)
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Controller/RestController.php]: Done.

File: src/Bpi/ApiBundle/Domain/Repository/FacetRepository.php (hunk 4 above)
Comment (rimi-itk) [inline src/Bpi/ApiBundle/Domain/Repository/FacetRepository.php]: Done.

**Pull request merged**